### PR TITLE
Added Automated Cert rotation feature

### PR DESCRIPTION
**Pull request by RaleighSF**
*Description of changes:*
This PR includes all of the necessary client modifications, YAML template changes, Lambda resources, policies and other automated activities needed to support custom certificate rotation for AWS IoT Core certificates (signed by AWS Root CA). Changes have been tested in the local account.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: SubTemplates/IoT/template.yaml [hunk 1]
```diff
@@ -69,15 +71,112 @@ Resources:
       Timeout: 10
       Policies:
         - AWSLambdaBasicExecutionRole
-  
+      Environment:
+        Variables:
+          ENV: dev
+      VpcConfig:
```
Comment (djjohnse) [inline SubTemplates/IoT/template.yaml]: This vpc config is blank and is causing errors for deploy

File: SubTemplates/IoT/Lambdas/provision_device/client/main.py [hunk 2]
```diff
@@ -1,49 +1,84 @@
 # ------------------------------------------------------------------------------
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-# SPDX-License-Identifier: MIT-0
+# SPDX-License-Identifier: MIT-0
```
Comment (djjohnse) [inline SubTemplates/IoT/Lambdas/provision_device/client/main.py:3]: I tried running with serial number. The client tried to create a thing with name 8675309 even though the serial number in machine_config was 12345ABCD. 

Connecting with official cert timed out

**Review by djjohnse — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -50,6 +51,16 @@ Completing the cloud setup dynamically creates an edge client for your account i
...
 
+### NEW! Cert Rotation 
+All things created in the AWS IoT registry will contain a *cert_issuance* attribute which will be the trigger for detecting certificates outside of the expiry date. A lambda trigger will run daily to detect expired certs and publish a message to each endpoint within the graceperiod of expiry. Your edge device must subscribe to this message and handle the published alert (arriving each day until compliance is met or certificate expires altogether). By executing the provided *run_provisioning* method (in main.py) and passing in run_provisioning(isRotation=True), the edge will perform a new cert rotation. The provisioning hook will validate if the cert_issuance date is indeed out of compliance and if so, issue a new certificate with an updated date (deactivating the old). The following illustrates a sample test workflow:
+
+1) Make sure to enable Fleet Index Settings (Thing Indexing) in IoT Core / Settings.
```
Comment (djjohnse) [inline README.md:57]: Can we make this change from CF?

**Comment by RaleighSF** [comment]
David, thanks for the review. These changes (like the VPC one) were made on my testing branch, but apparently weren't merged into the feature branch. I will reconcile my changes with the feature branch and issue a review. Appreciate the patience!

File: SubTemplates/IoT/template.yaml (hunk 1 above)
Comment (RaleighSF) [reply SubTemplates/IoT/template.yaml]: This was added in my testing branch but didn't make it in the official checkin. There now.

File: SubTemplates/IoT/Lambdas/provision_device/client/main.py (hunk 2 above)
Comment (RaleighSF) [reply SubTemplates/IoT/Lambdas/provision_device/client/main.py:3]: I originally took out the file based configuration for serial numbers but after some thought, added it back in. So your expected behavior should work in the latest checkin on the feature branch.

File: README.md (hunk 3 above)
Comment (RaleighSF) [reply README.md:57]: We can and it's not hard. I can make this a TODO but won't have time with recent client work for a week or two. Can I ask for a merge with the updated functionality to reconcile my workbench and clear my current train of thought ? I will ensure to come back and roll in the auto-fleet indexing.

**Comment by RaleighSF** [comment]
The latest check-in to the feature branch should have turn-key functionality for cert-rotation.

**Pull request merged**